### ISSUE TITLE
some questions about code

### ISSUE THREAD

**Issue by LSY2017** [jiahaoLjh/trajectory-pose-3d#4]
First,thank you for your excellent work.I have some questions about this project:
1.in 'dataset',there are:
self.left_right_symmetry_2d = np.array([0, 4, 5, 6, 1, 2, 3, 7, 8, 9, 10, 14, 15, 16, 11, 12, 13])
self.left_right_symmetry_3d = np.array([3, 4, 5, 0, 1, 2, 6, 7, 8, 9, 13, 14, 15, 10, 11, 12])

what do they mean?
In fact,I just want to predict 16 joints instead of 17,however these code make it difficult for me to do it.
2.the 'Transformer' part in code is this:
self.conv_feats = nn.ModuleList([
            nn.Conv1d((self.n_joints + 1) * 2, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 2, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 3, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 4, 256, 1, stride=1, padding=0),
        ])
right?
however I can't quite understand why Transformer is required.If we delete this part,or replace the conv1d operation with Linear,will the model still work?
sorry for my bad english,to express it clearly,i have to write it in chinese again

首先，非常感谢大佬的工作。对于这个项目我有一些问题：
1.我想把测试17个关键点改成测试16个关键点，我不需要human3.6m中的0号点。不过我有两行代码没看懂。dataset里有这两行：
self.left_right_symmetry_2d = np.array([0, 4, 5, 6, 1, 2, 3, 7, 8, 9, 10, 14, 15, 16, 11, 12, 13])
self.left_right_symmetry_3d = np.array([3, 4, 5, 0, 1, 2, 6, 7, 8, 9, 13, 14, 15, 10, 11, 12])
请问这两行是代表了啥意思？我有点找不到它们对应的是啥。
2.论文中的“Transformer”模块，在代码中是不是
self.conv_feats = nn.ModuleList([
            nn.Conv1d((self.n_joints + 1) * 2, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 2, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 3, 256, 1, stride=1, padding=0),
            nn.Conv1d((self.n_joints + 1) * 2 + 256 * 4, 256, 1, stride=1, padding=0),
        ])
这块？
我有点不太理解Transformer的作用。没有Transformer就不能work么？或者把Transformer中的Conv1d换成其它的操作，例如Linear，是否还能work呢？在iccv17的那篇论文中只有Linear也是能work的，这里的Conv1d是否有特殊的含义？

**Comment by jiahaoLjh** [comment]
@LSY2017 

1. The left_right_symmetry is used for data augmentation, i.e. introducing one more data for each data sample by switching the left and right limbs (arms and legs) for both input 2d and output 3d. The two lines indicate the corresponding joint indices after horizontal flipping for each joint. For example, 0 -> 0 (root), 1 -> 4 (right to left hip). You can refer to "dataset.py" for the list of joint indices.

2. No. The transformer is implemented with the following two lines
`transformed_feats = torch.matmul(feats, torch.transpose(bases, 1, 2)) / self.n_frames * 2`
`transformed_feats = transformed_feats.view(transformed_feats.shape[0], -1)  # (B, C, K)`
The codes you refer to are for mapping the input to high dimensional feature space.
